qa/tasks/cephfs/test_mirroring.py

```diff
@@ -1625,3 +1625,53 @@ def test_cephfs_mirror_remote_snap_corrupt_fails_synced_snapshot(self):
                    expected_snap_count == res[dir_name]['snaps_synced']):
                     break
         self.disable_mirroring(self.primary_fs_name, self.primary_fs_id)
+
+    def test_cephfs_mirror_sync_already_existing_snapshots(self):
```

Comment:
Isn't this a redundant test? I believe there is at least one test that checks if snapshots are synchronized as expected.

Reply:
No, this should be the only test that does `add_directory` after snapshot-ting. Is there any function does that already?

Reply:
But, how is this test relevant to this fix?

Reply:
@vshankar 
This test checks if the changes are taken from the previous snapshot or not, while adding a directory.
There are multiple things gets fixed by this PR.
1. Incremental sync after adding a directory for mirroring -> We have test for this.
2. Incremental sync of the already existing snapshots while adding a directory for mirroring -> We don't have test for this.
3. switch to remote dir_root when local prev snapshot is unavailable -> We have test for this.

Hope that clarifies.

Reply:
@vshankar I think it's better to have a separate test for it (add_directory after snapshot-ting), as the same thing is reported here: https://github.com/ceph/ceph/pull/54633#discussion_r1873219699.